### PR TITLE
Improve error messages for unresolved packages and projects

### PR DESCRIPTION
**Pull request by emgarten**
Adding new error codes and text for scenarios where packages cannot be found during restore.
* No versions of a package exist
* No stable versions of a package exist
* No versions fit the dependency range
* The project was not found

#### Package does not exist anywhere

```
NU1101: Unable to find package System.Missing. No packages exist with this id in source(s): dotnet-buildtools, dotnet-core, dotnet-roslyn, NuGet.org, NuGetVolatile
```

#### Package version is not within the range
```
NU1102: Unable to find package NuGet.Versioning with version (>= 9.0.1)
  - Found 30 version(s) in NuGet.org [ Nearest version: 4.0.0 ]
  - Found 10 version(s) in dotnet-buildtools [ Nearest version: 4.0.0-rc-2129 ]
  - Found 9 version(s) in NuGetVolatile [ Nearest version: 3.0.0-beta-00032 ]
  - Found 0 version(s) in dotnet-core
  - Found 0 version(s) in dotnet-roslyn
```
Fixes https://github.com/NuGet/Home/issues/2730

//cc @rrelyea @anangaur

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnresolvedMessages.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        /// Create a specific error message for the unresolved dependency.
+        /// </summary>
+        public static async Task<RestoreLogMessage> GetMessageAsync(IRestoreTargetGraph graph,
+            LibraryRange unresolved,
+            RemoteWalkContext context,
+            ILogger logger,
+            CancellationToken token)
+        {
+            string message = null;
+            var code = NuGetLogCode.NU1100;
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnresolvedMessages.cs]: Consider leaving a comment here. just for clarity.

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnresolvedMessages.cs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            else
+            {
+                // Unknown or non-specific.
+                // Also shown when no sources exist.
+                message = string.Format(CultureInfo.CurrentCulture,
+                    Strings.Log_UnresolvedDependency,
+                    unresolved.ToString(),
+                    graph.Name);
+
+                code = NuGetLogCode.NU1100;
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnresolvedMessages.cs]: same as above.

File: src/NuGet.Core/NuGet.Common/Errors/RestoreLogMessage.cs [hunk 3]
```diff
@@ -32,12 +32,12 @@ public class RestoreLogMessage : IRestoreLogMessage
             Message = errorString;
             Time = DateTimeOffset.UtcNow;
 
+            var graphList = new List<string>();
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Common/Errors/RestoreLogMessage.cs:35]: Can be changed to - 

```
            TargetGraphs = new List<string>();

            if (!string.IsNullOrEmpty(targetGraph))
            {
                TargetGraphs.Add(targetGraph);
            }
```

**Review by mishra14 — APPROVED** [review]
small changes. But looks good otherwise.

File: src/NuGet.Core/NuGet.Common/Errors/RestoreLogMessage.cs (hunk 3 above)
Comment (emgarten) [reply src/NuGet.Core/NuGet.Common/Errors/RestoreLogMessage.cs:35]: readonly

**Pull request merged**